internal/k8shandler/collection.go

```diff
@@ -199,17 +206,27 @@ func (clusterRequest *ClusterLoggingRequest) addSecurityLabelsToNamespace() erro
 	return nil
 }
 
-func (clusterRequest *ClusterLoggingRequest) RemoveInputServices() error {
+func (clusterRequest *ClusterLoggingRequest) RemoveInputServices(httpInputs []string, currOwner []metav1.OwnerReference, removeAllServices bool) error {
 	if clusterRequest.Cluster.Spec.Collection.Type != logging.LogCollectionTypeVector {
 		return nil
 	}
 
-	for _, input := range clusterRequest.Forwarder.Spec.Inputs {
-		if input.Receiver != nil && input.Receiver.HTTP != nil {
-			if err := clusterRequest.RemoveInputService(input.Name); err != nil {
+	// Find all http input services by label
+	labelSelector, _ := labels.Parse("component=http-input-service")
```

Comment:
```
labelSelector, _ := labels.Parse("component=http-input-service")
&client.ListOptions{LabelSelector: labelSelector}
```
can be lifted to the package level as they don't depend on anything inside this function.